frontend/app/assets/javascripts/spree/frontend/checkout.js

```diff
@@ -61,15 +61,17 @@ Spree.ready(function () {
 })
 
 $('.js-remove-credit-card').click(function() {
-  return $.ajax({
-    async: false,
-    method: 'DELETE',
-    url: Spree.routes.api_v2_storefront_destroy_credit_card(this.dataset.id),
-    dataType: 'json',
-    headers: {
-      'Authorization': 'Bearer ' + SpreeAPI.oauthToken
-    }
-  }).done(function() {
-    location.reload();
-  })
+  if(confirm('Are you sure you want to delete the credit card?')) {
```

Comment:
```suggestion
  if(confirm('Are you sure you want to delete the credit card?')) {
```

let's grab that text from translation please :) you can add it here: https://github.com/spree/spree/blob/master/frontend/app/views/spree/shared/_translations.html.erb